### PR TITLE
feat: api metrics

### LINKED ISSUE #56: SuperHot HTTP API metrics
TBD

### PR DESCRIPTION
**Pull request by vasco-santos**
This PR adds metrics endpoint to the API by:
- new table for metrics
- cron job to compute them async
- use cron job released

From previous knowledge doing the queries in the API itself would work in the beginning. But let's not wait for fires, and go straight for cron job

Blocked on https://github.com/nftstorage/nftstorage.link/pull/86

Closes #56

### CODE REVIEW

File: packages/api/src/metrics.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    `nftlinkapi_permacache_urls_total ${urlsTotal}`,
+    `# HELP nftlinkapi_permacache_users_total Total number of users with perma cached urls.`,
+    `# TYPE nftlinkapi_permacache_users_total counter`,
+    `nftlinkapi_permacache_users_total ${usersTotal}`,
+    `# HELP nftlinkapi_permacache_size_total Total perma cached size.`,
+    `# TYPE nftlinkapi_permacache_size_total counter`,
+    `nftlinkapi_permacache_size_total ${sizeTotal}`,
+    `# HELP nftlinkapi_permacache_events_total Total perma cache events.`,
+    `# TYPE nftlinkapi_permacache_events_total counter`,
+    `nftlinkapi_permacache_events_total ${eventsTotal}`,
```
Comment (alanshaw) [inline packages/api/src/metrics.js:39]: Split by type?

```suggestion
    `nftlinkapi_permacache_events_total{type="Put"} ${putEventsTotal}`,
    `nftlinkapi_permacache_events_total{type="Delete"} ${deleteEventsTotal}`,
```

File: packages/cron/src/jobs/metrics.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+ * @typedef {import('pg').Pool} Client
+ * @typedef {{ name: string, value: number }} Metric
+ */
+
+const COUNT_USERS = `
+SELECT COUNT(*) AS total
+FROM nftstorage.user u JOIN nftstorage.user_tag ut
+ON u.id = ut.user_id
+WHERE ut.tag = 'HasSuperHotAccess'::user_tag_type AND ut.value = 'true'
+`
```
Comment (alanshaw) [inline packages/cron/src/jobs/metrics.js:17]: Isn't this just a count of `HasSuperHotAccess` tags?

```suggestion
const COUNT_USERS = `
SELECT COUNT(*) AS total
FROM nftstorage.user_tag
WHERE tag = 'HasSuperHotAccess'::user_tag_type AND value = 'true'
`
```

**Review by alanshaw — APPROVED** [review]
LGTM, added some non-blocking suggestions.

File: packages/api/src/metrics.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  ].join('\n')
```
[the remark this answers is not in the document]
Comment (vasco-santos) [reply packages/api/src/metrics.js:43]: I will do this as follow up.

**Pull request merged**